js/flightlog_parser.js

```diff
@@ -582,23 +582,20 @@ var FlightLogParser = function(logData) {
                 // Extract the firmware revision in case of Betaflight/Raceflight/Cleanfligh 2.x/Other
                 var matches = fieldValue.match(/(.*flight).* (\d+)\.(\d+)(\.(\d+))*/i);
                 if(matches!=null) {
-                
-                    // The that.sysConfig.firmwareType usually has been detected before that. Default value.
-                    if (typeof that.sysConfig.firmwareType === 'undefined' || !that.sysConfig.firmwareType) {
-                	    that.sysConfig.firmwareType = FIRMWARE_TYPE_BETAFLIGHT;                    
+
+                    // Detecting Betaflight requires looking at the revision string
+                    if (matches[1] === "Betaflight") {
```

Comment:
I don't know if it can break compatibility with very old logs (I wasn't here ;) ).
The previous code, assigns the BETAFLIGHT firmware by default when the firmware type is not informed (is undefined or null). I don't know if there were some logs some time ago with "Firmware revision" but not "Firmware type". I don't think that exists such kind of logs but maybe someone can told us.